### PR TITLE
Tweak convert

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR updates the convert to always output to `temp_echopype_output` when save_path is None. This also updates the tests to use local services.

NOTE: This is exactly the same as #257.

### CODE REVIEW

File: echopype/convert/convert.py [hunk 1]
```diff
@@ -242,27 +242,19 @@ def _validate_path(self, file_format, save_path=None):
...
-            else:
-                # Default output directory taken from first input file
-                out_dir = Path(fsmap.root).parent.absolute()
-
-                # Check permission, raise exception if no permission
-                io.check_file_permissions(out_dir)
+            current_dir = Path.cwd()
+            # Check permission, raise exception if no permission
+            io.check_file_permissions(current_dir)
+            out_dir = current_dir.joinpath(Path("temp_echopype_output"))
```
Comment (lsetiawan) [inline echopype/convert/convert.py:249]: This satisfy: #251

**Comment by emiliom** [comment]
Woo-hoo! 
I'll review it over the weekend, hopefully tomorrow.

**Comment by lsetiawan** [comment]
Thanks @emiliom! Have a great weekend! Stay warm :smile:

**Comment by lsetiawan** [comment]
Ping @emiliom, What is the status of this review?

**Comment by emiliom** [comment]
Sorry for the long delay :disappointed: . My goal is to finally get to it late this morning. Thanks for checking in.

**Comment by lsetiawan** [comment]
Sounds good. No worries :smile: Thank you!

**Comment by emiliom** [comment]
@lsetiawan  I'm at a loss to figure out how to run the comprehensive suite of tests locally, that fully use this infrastructure. Previously, I was only testing http I/O, so I followed your GH actions instructions to build & deploy the former `.ci_helpers/docker/http.dockerfile` (that no longer exists), and disabled all the minio tests. Now there is a stack of 3 docker-based set ups that need to happen before I can run `test_convert.py`: `.github/actions/gdrive-rclone`, `minioci`, and `httpd:2.4`. I'm not entirely sure how to set up this local dev environment. I see the docker compose file `.ci_helpers/docker/docker-compose.yaml` that seems to build both minio and httpd and says specifically in a comment line: "For local development only to spin up services to test against".

I can try to spin that up with docker compose. But what about `gdrive-rclone`? Is it via `.github/actions/gdrive-rclone/Dockerfile`?  But I need to pass credentials to it, no?

Brief instructions would be helpful. Once I've fully reproduced things locally, I can add and flesh out those instructions into the echopype documentation, to ease onboarding of contributors -- including our own work.

**Comment by lsetiawan** [comment]
@emiliom I added the gdrive-rclone into the docker-compose now :smile: And see below for instructions.

## Testing Instructions

1. Go to the root directory of repo.
2. Set environmental variables

    ```bash
    export GOOGLE_SERVICE_JSON=$(cat /path/to/google-creds.json)
    export TEST_DATA_FOLDER_ID=1gnXbrNCT3BO6QM2Ro09kCyXztSe99uzc
    ```

3. Build the docker compose images

    ```bash
    docker-compose -f .ci_helpers/docker/docker-compose.yaml build
    ```

4. Bring up all the docker compose services

    ```bash
    docker-compose -f .ci_helpers/docker/docker-compose.yaml up
    ```

5. Copy test_data into http server

    ```bash
    docker cp -L ./echopype/test_data docker_httpserver_1:/usr/local/apache2/htdocs/data
    ```

6. Run the test :) (Assuming you've done all the installs locally for echopype!)

    ```bash
    python -m pytest --log-cli-level=WARNING --verbose echopype/tests/test_convert.py
    ```

**Comment by emiliom** [comment]
Thank you @lsetiawan !! That seems crystal clear; you're making me feel bad that I may not have to figure anything out by myself :wink:

I'm hoping to get back to this late in the afternoon.

**Comment by lsetiawan** [comment]
Haha no problem at all @emiliom! I didn't think about the github actions piece locally before :stuck_out_tongue:

**Comment by emiliom** [comment]
Next question: How do I create `google-creds.json`?

**Comment by lsetiawan** [comment]
Sorry for the delayed response, I just sent you the credential json file. I totally forgot to send that over :sweat_smile:

**Comment by emiliom** [comment]
Thanks! I'll try the whole thing in an hour.

**Comment by emiliom** [comment]
@lsetiawan I'm getting there. But I've run into this error:

```bash
$ docker-compose -f .ci_helpers/docker/docker-compose.yaml build
httpserver uses an image, skipping
Building minio
Step 1/3 : FROM minio/minio
 ---> 7ea4a619ecfc
Step 2/3 : RUN microdnf install git
 ---> Running in 4c5257f6b515
/bin/sh: microdnf: not found
ERROR: Service 'minio' failed to build: The command '/bin/sh -c microdnf install git' returned a non-zero code: 127
```

**Comment by emiliom** [comment]
Ok, I've resolved this by adding `FROM registry.fedoraproject.org/fedora-minimal:30` to the `.ci_helpers/docker/minioci.dockerfile`.

I'm seeing this warning in the docker compose build:
```bash
(microdnf:1): libdnf-WARNING **: 20:50:28.231: Loading "/etc/dnf/dnf.conf": IniParser: Can't open file
```

But we'll see if it amounts to anything.

**Comment by emiliom** [comment]
And now on docker compose `up`, I ran into this **minio** error:

```bash
Creating network "docker_echopypenetwork" with the default driver
Creating network "docker_default" with the default driver
Creating docker_rclone_1 ... done
Creating docker_minio_1      ... error
Creating docker_httpserver_1 ... 

ERROR: for docker_minio_1  Cannot start service minio: OCI runtime create failed: container_linux.go:349: staCreating docker_httpserver_1 ... done

ERROR: for minio  Cannot start service minio: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"minio\": executable file not found in $PATH": unknown
ERROR: Encountered errors while bringing up the project.
```

I took a quick look to see if I could fix this, but I think I'll just leave it to @lsetiawan . Thanks! Getting there.

**Comment by emiliom** [comment]
Don helped me through this problem. Ultimately, it had to do with an old `minion/minion` docker image I had from a while ago. Removing it fixed the docker compose up problem. The `test_convert.py` completed successfully!

**Comment by emiliom** [comment]
@lsetiawan I noticed that the `data` folder that is created on the local http server (`http://localhost:8080/data/`) has files that are not used in `test_convert.py`, last time I checked. That's all the `from_matlab`, `from_echoview` and `ncei-wcsd` sub-folders, and other miscellaneous files. That's all being downloaded (rclone'd) by `gdrive-rclone`, w/o any real need. Now, it's the actual tests that take most of the time, by far; but still, wouldn't it be better to remove those files from the Google Drive folder?

I'm done with echopype for the day. I'll dig into the tests tomorrow. Thanks again @lsetiawan !

**Comment by lsetiawan** [comment]
We can remove those files, but I'm not sure if they will be needed for later tests. I think that's a whole separate discussion :grimacing:

**Review by emiliom — APPROVED** [review]
Woo-hoo!

**Comment by emiliom** [comment]
> We can remove those files, but I'm not sure if they will be needed for later tests. I think that's a whole separate discussion

Agreed, so I've created a new issue (#263) that we can refer back to in the future. Now I'll merge this PR, finally :partying_face:

**Pull request merged**